core/src/agent/state.rs

```diff
@@ -194,13 +192,20 @@ fn reduce_get_entry(
 ) {
     let action = action_wrapper.action();
     let address = unwrap_to!(action => Action::GetEntry);
+    let storage = &state.chain().content_storage().clone();
+    let json = storage
+        .read()
+        .unwrap()
+        .fetch(&address)
+        .expect("could not fetch from CAS");
+    let result: Option<SerializedEntry> = match json {
+        Some(js) => match js.try_into() {
+            Ok(info) => Some(info),
```

Comment:
Not really important, but why `info`? This is the SerializedEntry, no?

Reply:
oh that is a typo, i think I meant to type "into" which is not a really good name in itself haha. I will change that!

Reply:
Also, you can replace this `match` statement with a call to `.ok()`

Reply:
it converts a `Result<T, E>` to a `Option<T>`

Reply:
Oh yes! That is a good idea! Will make those changes! @sphinxc0re

Reply:
```rust
let result: Option<SerializedEntry> = json.and_then(|js| js.try_into().ok())
```